ui/apps/platform/src/Containers/MainPage/Body.tsx

```diff
@@ -83,19 +92,152 @@ const AsyncSystemHealthPagePF = asyncComponent(
     () => import('Containers/SystemHealth/PatternFly/SystemHealthDashboard')
 );
 
+/*
+ * basePath like violationsBasePath = '/main/violations' is key of routeDescriptorMap in routePaths.ts
+ * propPath includes parameters like path="/main/violations/:alertId?"
+ */
+type BaseRouteComponent = {
+    basePath: string;
```

Comment:
Regarding your number 6 residue, you could potentially get the best of both worlds by defining a form of type alias for a `RoutePath` and use that as the type for `basePath` here and as the Record key in `routeDescriptorMap`.

```typescript
export type RoutePath = string & { readonly tag: unique symbol };

// Use type assertions where the routes are defined
export const dashboardPath = `/main/dashboard` as RoutePath;

// Or use an un-exported constructor function to prevent outside modules from creating `RoutePaths` without a type assertion
const makeRoutePath = (s: string) => s as RoutePath;

function testAsRoutePath(t: RoutePath) {}
function testAsString(s: string) {}

testAsRoutePath(dashboardPath); // ok
testAsRoutePath('regular string'); // type error
testAsString(dashboardPath); // ok
testAsString('regular string'); // ok
```

I'm not sure this is necessarily the best approach as it feels a little obscure, but it would give you some type safety without the need to duplicate each route.

Reply:
Wow, super solution to annotate with `as RoutePath` and distinguish from the other paths.